prometheus_juju_exporter/collector.py

```diff
@@ -11,8 +13,26 @@ def __init__(self, registry=None, debug=False):
         """Create new collector daemon and configure runtime environment."""
         self.config = Config().get_config()
         self.logger = get_logger(debug=debug)
+        self.controller = Controller()
+        self.model = Model()
         self._registry = registry or CollectorRegistry()
         self.metrics = {}
+        self.currently_cached_labels = {}
+        self.previously_cached_labels = {}
+        self.logger.debug("Collector initialized")
+        self.prefixes = [
```

Comment:
I think these should rather live in the config file, preferably as a mapping between addresses and vendor.

Reply:
I agree that the current approach to store prefixes is not perfect, but moving it to config file would require both charm and snap to add it as an config option. I think this change fits more in the `allow mac address detection overrides` task which is separate from the first iteration.